remme/rpc_api/event/_handlers.py

```diff
@@ -134,14 +134,30 @@ class BlockEventHandler(BaseEventHandler):
     def hash_keys(cls):
         return ('id',)
 
-    def prepare_response(self, state, validated_data):
-        return {'id': state}
+    async def prepare_response(self, state, validated_data):
+        """
+        Prepare response.
+
+        To fetch block by number, consider blocks like elements in array. To get second, count from zero.
+        """
+        block_information_client = BlockInfoClient()
+        block_information = await block_information_client.get_block_info(block_num=int(state.get('block_num')) - 1)
```

Comment:
What will be in case when we do not have 'block_num' ?

Reply:
@DeV1doR this is actually impossible. Block validation rules does not allow to have blocks without `BlockInfo` except for genesis block which will never ever appear in block events.

Reply:
Than use `state['block_num']`

Reply:
@DeV1doR, because you safely can get the vale by key? Thinks, [this](https://github.com/Remmeauth/remme-core/pull/189#discussion_r246409249) is more important.

Reply:
@dmytrostriletskyi if it not safe, why are u sure, that this `int(state.get('block_num'))` will not raise `TypeError` ?

Reply:
@DeV1doR, done.